fvm debug execution

vyzo opened the pull request:
Companion to https://github.com/filecoin-project/ref-fvm/pull/605

raulk commented on rust/src/fvm/machine.rs:
```diff
@@ -132,6 +132,84 @@ fn create_fvm_machine(
 }
 
+#[ffi_export]
+fn create_fvm_debug_machine(
```
Can we refactor this to avoid duplicating the whole function?

vyzo replied:
sure, once it provably works :)

vyzo replied:
done.

raulk commented:
@vyzo Can you please remove the duplication so we can merge this PR?

vyzo commented:
yes of course!

vyzo commented:
Changes:
- rebased on master and squashed a few commits
- patched ref-fvm to point to `release/v1` branch which contains the debug execution feature
- deduplicated the code.

Should be ready now.

raulk reviewed (commented):
LGTM besides the comments. Would prefer @Stebalien to give the final signoff as a more active maintainer of this repo.

vyzo commented on rust/src/fvm/machine.rs:
```diff
@@ -80,19 +78,20 @@ fn create_fvm_machine(
-                Some(cid)
-            } else {
+            let manifest_cid = match manifest_cid {
+                Some(manifest_cid) => {
+                    let cid = Cid::try_from(&manifest_cid[..])
+                        .map_err(|err| anyhow!("invalid manifest: {}", err))?;
+                    Some(cid)
+                }
                 // handle cid.Undef for no manifest
                 // this can mean two things:
                 // - for pre nv16, use the builtin bundles
                 // - for nv16 or higher, it means we have already migrated state for system
                 //   actor and we can pass None to the machine constructor to fish it from state.
                 // The presence of the manifest cid argument allows us to test with new bundles
                 // with minimum friction.
-                None
+                None => None,
```
map didnt like me, but i can temove the comment completely.

vyzo replied:
it came from outer space... this language is an obscurity competition.

vyzo commented on rust/src/fvm/machine.rs:
```diff
@@ -102,9 +101,28 @@ fn create_fvm_machine(
                     network_config.override_actors(manifest);
                 }
-                Ok(None) => {}
+                Ok(None) => (),
```
its equivalent, so its a matter of style; fine with reverting it.

Stebalien reviewed (approved):
This works (once we've removed the branch dependencies). Are we planning on getting this into the lotus release? If so, we should revert the dependency updates in Cargo.toml.

vyzo commented:
I think we are aiming for lotus master -- @arajasek ?
